Source/Nalu.Maui.Navigation/NaluShell.cs

```diff
@@ -98,7 +98,40 @@ protected override void OnHandlerChanged()
     }
 
     /// <inheritdoc />
-    protected override async void OnNavigating(ShellNavigatingEventArgs args)
+    protected override bool OnBackButtonPressed()
```

Comment:
This is basically the same code found in the base class `Shell.OnBackButtonPressed` with the following code replaced by Nalu pop navigation.

```
var currentContent = CurrentItem?.CurrentItem;
if (currentContent != null && currentContent.Stack.Count > 1)
{
    NavigationPop();
    return true;
}
```

https://github.com/dotnet/maui/blob/0893ce0955144358421f41866a69edc9479e8c2a/src/Controls/src/Core/Shell/Shell.cs#L1535-L1540

The huge problem here is that `NavigationPop` is triggering navigation via `Navigation` and not via `Shell` itself so it's not possible to hook into that.